### DISCUSSION TITLE
Testing incoming email in dev

### DISCUSSION THREAD

**Discussion by ska-ibees** [Tecnativa/doodba#404] [Q&A] [answered]
Hello,

With MailHog, we can locally test outgoing and also we can release any message to external SMTP and this flow is working fine. However, it seems that replying to the messages is not supported by mailhog.

Any idea how to test incoming emails/replies in the dev env?

**Comment by pedrobaeza** [comment, marked as answer]
The only way right now is to change devel.yaml definition for not being isolated (internal: false), and add an incoming mail server for that. Not sure if we can add any mechanism for "injecting" incoming mails. Maybe a local squirrelmail or similar? @Yajo @joao-p-marques

**Reply by ska-ibees** [reply]
Thanks again for your continued great support!

**Reply by yajo** [reply]
The only way to have a completely local mail I/O gateway would be to... set it up! 😆 

There are a few tools that should help you do that:

- https://github.com/antespi/docker-imap-devel
- https://github.com/Mailu/Mailu
- https://github.com/docker-mailserver/docker-mailserver

Once done, you should be able to point your mail client to localhost and use it, although it could still have some extra quirks to do due to how DNS is so deeply involved in mail servers.

Possibly the easiest way to do it is to use a real address, like @pedrobaeza said.

**Reply by ska-ibees** [reply]
Thanks, Yajo. Mr Pedro's reply solved my issue as I am working on a private project and can test email functionality with the real email addresses. However, many other folks may find your answer helpful.
